.yarn/patches/@lavamoat-lavapack-npm-3.3.0-5a61b5374d.patch

```diff
@@ -0,0 +1,55 @@
+diff --git a/src/runtime.js b/src/runtime.js
+index 97c9ddb9b76a2fe31cf9b202cb491f15232232ff..02d657e84d6d4572d9b6eb5e73caa01380c89b0c 100644
+--- a/src/runtime.js
++++ b/src/runtime.js
+@@ -10831,6 +10831,10 @@ function makeGetEndowmentsForConfig ({ createFunctionWrapper }) {
+ function getPropertyDescriptorDeep (target, key) {
+   let receiver = target
+   while (true) {
++    // abort if this is the end of the prototype chain.
++    if (!receiver) {
++      return { prop: null, receiver: null }
++    }
+     // support lookup on objects and primitives
+     const typeofReceiver = typeof receiver
+     if (typeofReceiver === 'object' || typeofReceiver === 'function') {
+@@ -10845,10 +10849,6 @@ function getPropertyDescriptorDeep (target, key) {
+       // eslint-disable-next-line no-proto
+       receiver = receiver.__proto__
+     }
+-    // abort if this is the end of the prototype chain.
+-    if (!receiver) {
+-      return { prop: null, receiver: null }
+-    }
+   }
+ }
+ 
+@@ -11384,7 +11384,9 @@ module.exports = {
+         const descriptors = Object.getOwnPropertyDescriptors(source)
+         Object.values(descriptors).forEach(desc => {
+           if ('get' in desc) {
++            try {
+             Reflect.apply(desc.get, globalRef, [])
++            }catch{}
+           }
+         })
+       })
+@@ -11556,6 +11558,18 @@ module.exports = {
+ 
+ 
+   function loadModuleData (moduleId) {
++    if (typeof window === 'undefined' && require('resolve').isCore(moduleId)) {
```

Comment:
there are modules, installable from npm, that have the same names as builtins. This would have to be a bit more careful if we were to upstream it.

Reply:
This is quite similar to the way LavaMoat Node handles it: https://github.com/LavaMoat/LavaMoat/blob/main/packages/node/src/kernel.js#L81

I was under the impression that NPM modules would be included in the bundle and thus referenced by a number as their `moduleId`. But I do not have the entire context 😅

Reply:
if you assume `moduleId` is a number, why do you believe `require('resolve').isCore(moduleId)` works?

I don't have full context here either, but every module is identified by its specifier at some point and it's a matter of looking things up in a specific order to prioritize local dependencies over builtins. Also, figuring out when that's desired and when not is a problem now. For a frontend bundle the default is to bundle up the builtin shims and for running in node you need to actively avoid bundling them even if they exist in the dependency tree. 
We've arrived at target platform config for bundler.

Reply:
> if you assume moduleId is a number, why do you believe require('resolve').isCore(moduleId) works?

Sorry, I meant for builtins it seems that `moduleId` is a string, while for everything else that references other modules in the bundle, a number is used.